tests/jsunit/xml_test.js

```diff
@@ -131,31 +145,137 @@ function test_domToText() {
+  // Expect that workspace still loads without serialized variables.
+  Blockly.defineBlocksWithJsonArray([{
+    'type': 'field_variable_test_block',
+    'message0': '%1',
+    'args0': [
+      {
+        'type': 'field_variable',
+        'name': 'VAR',
+        'variable': 'item'
+      }
+    ],
+  }]);
+  xmlTest_setUpWithMockBlocks();
+  var mockGenUid = mockControl_.createMethodMock(Blockly.utils, 'genUid');
+  mockGenUid().$returns('1');
+  mockGenUid().$returns('1');
+  mockGenUid().$replay();
   try {
     var dom = Blockly.Xml.textToDom(
         '<xml xmlns="http://www.w3.org/1999/xhtml">' +
-        '  <block type="test_block" inline="true" x="21" y="23">' +
+        '  <block type="field_variable_test_block" id="block_id"' +
+        ' x="263" y="212">' +
+        '    <field name="VAR">name1</field>' +
+        '  </block>' +
+        '</xml>');
+    Blockly.Xml.domToWorkspace(dom, workspace);
+    assertEquals('Block count', 1, workspace.getAllBlocks().length);
+    xmlTest_checkVariableValues('name1', '', '1');
+  } finally {
+    delete Blockly.Blocks.test_block;
+    xmlTest_tearDownWithMockBlocks();
+  }
+}
+function test_domToWorkspace_VariablesAtTop() {
+  // Expect that unused variables are preserved.
+  Blockly.defineBlocksWithJsonArray([{
+    'type': 'field_variable_test_block',
+    'message0': '%1',
+    'args0': [
+      {
+        'type': 'field_variable',
+        'name': 'VAR',
+        'variable': 'item'
+      }
+    ],
+  }]);
+  xmlTest_setUpWithMockBlocks();
+  try {
+    var dom = Blockly.Xml.textToDom(
+        '<xml xmlns="http://www.w3.org/1999/xhtml">' +
+        '  <variables>' +
+        '    <variable type="type1" id="id1">name1</variable>' +
+        '    <variable type="type2" id="id2">name2</variable>' +
+        '    <variable type="" id="id3">name3</variable>' +
+        '  </variables>' +
+        '  <block type="field_variable_test_block" id="block_id"' +
+        ' x="263" y="212">' +
+        '    <field name="VAR" id="id3" variabletype="">name3</field>' +
         '  </block>' +
         '</xml>');
     Blockly.Xml.domToWorkspace(dom, workspace);
     assertEquals('Block count', 1, workspace.getAllBlocks().length);
+    xmlTest_checkVariableValues('name1', 'type1', 'id1');
+    xmlTest_checkVariableValues('name2', 'type2', 'id2');
+    xmlTest_checkVariableValues('name3', '', 'id3');
   } finally {
     delete Blockly.Blocks.test_block;
+    xmlTest_tearDownWithMockBlocks();
+  }
+}
 
+function test_domToWorkspace_VariablesAtTop_MissingType() {
+  // Expect thrown error when a variable tag is missing the type attribute.
+  workspace = new Blockly.Workspace();
+  try {
+    var dom = Blockly.Xml.textToDom(
+        '<xml xmlns="http://www.w3.org/1999/xhtml">' +
+        '  <variables>' +
+        '    <variable id="id1">name1</variable>' +
+        '  </variables>' +
+        '  <block type="field_variable_test_block" id="e7/%Uw;F4fQ=9-n@V4u%"' +
```

Comment:
For succinctness, get rid of the block id and x and y.

In general in your tests, use human-readable IDs instead of random ones.  If you don't need the ID for the test strip it out, and if you don't need block positions strip those out as well.